borg/xattr.py

```diff
@@ -205,7 +205,7 @@ def func(path, buf, size):
 
         n, buf = _listxattr_inner(func, path)
         return [os.fsdecode(name) for name in split_string0(buf[:n])
-                if not name.startswith(b'system.posix_acl_')]
+                if name and not name.startswith(b'system.posix_acl_')]
```

Comment:
there is no spec.

on linux it's invalid, because names have to be qualified (`namespace.attr`). i don't know about osx. freebsd is nebolous, because "names may/may not be defined". the kernel code handles attrname==NULL specially. so if that's a valid use case then it was broken before as well, because empty string != NULL.